src/utils/print-utils.test.ts

```diff
@@ -34,40 +34,43 @@ describe(printValue, () => {
 
 describe(bracketsIfNeeded, () => {
     test.each`
-        input                         | output
-        ${'a'}                        | ${'a'}
-        ${'a b'}                      | ${'(a b)'}
-        ${'func(example ok)'}         | ${'func(example ok)'}
-        ${'func (example ok)'}        | ${'(func (example ok))'}
-        ${'<a'}                       | ${'(<a)'}
-        ${'<<a>'}                     | ${'(<<a>)'}
-        ${'<<a>>'}                    | ${'<<a>>'}
-        ${'{ prop: value }'}          | ${'{ prop: value }'}
-        ${'{ prop: value }.autoCast'} | ${'({ prop: value }.autoCast)'}
-        ${'{a} {b}'}                  | ${'({a} {b})'}
-        ${'"{a} {b}"'}                | ${'"{a} {b}"'}
-        ${'"{a} \\" {b}"'}            | ${'"{a} \\" {b}"'}
-        ${'"{a} {b}'}                 | ${'("{a} {b})'}
+        input                          | output
+        ${'a'}                         | ${'a'}
+        ${'a b'}                       | ${'(a b)'}
+        ${'func(example ok)'}          | ${'func(example ok)'}
+        ${'func (example ok)'}         | ${'(func (example ok))'}
+        ${'<a'}                        | ${'(<a)'}
+        ${'<<a>'}                      | ${'(<<a>)'}
+        ${'<<a>>'}                     | ${'<<a>>'}
+        ${'{ prop: value }'}           | ${'{ prop: value }'}
+        ${'{ prop: value }.autoCast'}  | ${'({ prop: value }.autoCast)'}
+        ${'AutoCast<{ prop: value }>'} | ${'AutoCast<{ prop: value }>'}
+        ${'{a} {b}'}                   | ${'({a} {b})'}
+        ${'"{a} {b}"'}                 | ${'"{a} {b}"'}
+        ${'"{a} \\" {b}"'}             | ${'"{a} \\" {b}"'}
+        ${'"{a} {b}'}                  | ${'("{a} {b})'}
     `('bracketsIfNeeded($input) => $output', ({ input, output }) => {
         expect(bracketsIfNeeded(input)).toBe(output);
     });
 
     test.each`
-        input      | output
-        ${'a'}     | ${'a'}
-        ${'a b'}   | ${'(a b)'}
-        ${'a & b'} | ${'a & b'}
-        ${'a | b'} | ${'(a | b)'}
+        input       | output
+        ${'a'}      | ${'a'}
+        ${'a b'}    | ${'(a b)'}
+        ${'a & b'}  | ${'a & b'}
+        ${'a | b'}  | ${'(a | b)'}
+        ${'a &| b'} | ${'(a &| b)'}
     `('bracketsIfNeeded($input, "&") => $output', ({ input, output }) => {
         expect(bracketsIfNeeded(input, '&')).toBe(output);
     });
 
     test.each`
-        input      | output
-        ${'a'}     | ${'a'}
-        ${'a b'}   | ${'(a b)'}
-        ${'a & b'} | ${'(a & b)'}
-        ${'a | b'} | ${'a | b'}
+        input       | output
+        ${'a'}      | ${'a'}
+        ${'a b'}    | ${'(a b)'}
+        ${'a & b'}  | ${'(a & b)'}
+        ${'a |& b'} | ${'(a |& b)'}
```

Comment:
Do we need to include a `test.each` where `bracketsIfNeeded` is called with both `|` and `&`, now that that's possible?